packages/waku/src/lib/renderers/html.ts

```diff
@@ -164,14 +164,15 @@ export async function renderHtml(
         Omit<ComponentProps<typeof INTERNAL_ServerRoot>, 'children'>
       >,
       { elementsPromise },
+      ...headElements.filter((ele) => ele.props?.async !== true),
       // See: https://github.com/wakujs/waku/pull/1545
       // isolate `React.use` in its own component
       // https://github.com/facebook/react/issues/33937#issuecomment-3091349011
       createElement(async () => {
         const resolvedHtmlNode = await htmlNode;
         return createElement(HtmlNodeWrapper, null, resolvedHtmlNode);
       }),
-      ...headElements,
+      ...headElements.filter((ele) => ele.props?.async === true),
```

Comment:
why are these added separately?

Reply:
It seems like some head elements should come before to avoid FOUC and some should come after to avoid hydration mismatch errors.